File: 3381-shortest-subarray-with-or-at-least-k-i/3381-shortest-subarray-with-or-at-least-k-i.java
class Solution {
    public int minimumSubarrayLength(int[] nums, int k) {
        if(nums.length<=0 || k<0) return -1;
        int ans=Integer.MAX_VALUE;
        for(int i=0;i<nums.length;i++){
            int v=0;
            for(int j=i;j<nums.length;j++){
                v|=nums[j];
                if(v>=k){
                    ans=Math.min(ans,j-i+1);
                    break;
                }
            }
            
        }
        return ans==Integer.MAX_VALUE?-1:ans;
    }
}

File: 3381-shortest-subarray-with-or-at-least-k-i/3381-shortest-subarray-with-or-at-least-k-i.cpp
class Solution {
public:
    int minimumSubarrayLength(vector<int>& nums, int k) {
        if(nums.size()<=0 || k<0) return -1;
        int ans=INT_MAX;
        for(int i=0;i<nums.size();i++){
            int v=0;
            for(int j=i;j<nums.size();j++){
                v|=nums[j];
                if(v>=k){
                    ans=min(ans,j-i+1);
                    break;
                }
            }
            
        }
        return ans==INT_MAX?-1:ans;
    }
};